src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -8,50 +8,91 @@
     read_product_team,
 )
 from domain.core.cpm_product.v1 import CpmProduct
-from domain.core.device.v3 import Device
-from domain.core.error import InvalidSpineMhsResponse
+from domain.core.device.v3 import (
+    MHS_DEVICE_NAME,
+    Device,
+    DeviceKeyAddedEvent,
+    DeviceReferenceDataIdAddedEvent,
+    DeviceTagAddedEvent,
+    QuestionnaireResponseUpdatedEvent,
+)
+from domain.core.device_key.v1 import DeviceKeyType
+from domain.core.device_reference_data.v1 import DeviceReferenceData
+from domain.core.error import ConfigurationError
+from domain.core.product_team.v3 import ProductTeam
 from domain.core.questionnaire.v3 import Questionnaire, QuestionnaireResponse
+from domain.repository.device_reference_data_repository.v1 import (
+    DeviceReferenceDataRepository,
+)
 from domain.repository.device_repository.v3 import DeviceRepository
 from domain.repository.questionnaire_repository.v2 import QuestionnaireRepository
 from domain.repository.questionnaire_repository.v2.questionnaires import (
     QuestionnaireInstance,
 )
-from domain.request_models.v1 import CreateMhsDeviceIncomingParams
+from domain.request_models.v1 import CpmProductPathParams, CreateMhsDeviceIncomingParams
 from domain.response.validation_errors import mark_validation_errors_as_inbound
 
 
 @mark_validation_errors_as_inbound
-def parse_mhs_device_payload(data, cache) -> Device:
+def parse_mhs_device_payload(data, cache) -> CreateMhsDeviceIncomingParams:
     payload: dict = data[parse_event_body]
     return CreateMhsDeviceIncomingParams(**payload)
 
 
+def check_for_existing_mhs(data, cache):
```

Comment:
Still not sure about this. I don't see it on the diagram or in the ACs that it can only have one MHS_DEVICE.